### DISCUSSION TITLE
How are fileworlds parsed

### DISCUSSION THREAD

**Discussion by chakalakasp** [d8ahazard/sd_dreambooth_extension#405] [General]
A big foggy on filewords.  I see the dropdown format settings -- my questions are:

What is the proper form of a file?  Say for instance one wanted to go "instance+class+description".  Would that look like:

jgowhg person A man kicks a soccerball.png
or
jgowhg+person+A man kicks a soccerball.png

Or does it not matter?

Second, does the text encoder in any way parse the description, or just the instance and class?  Does it help the training at all to put in the description that it's a man kicking a soccer ball or could the description be anything to make the filename unique and the encoder could care less?  If it helps I can rename files or make TXT sidecars or whatever, but I don't want to put in tons of metadata if the encoder doesn't care.

**Comment by wktra** [comment]
There's no tutorial on this, no help, no guide. Filewords is a mystery as far as I'm concerned. @chakalakasp 

So I might as drop this here to let you know how DIFFICULT it is to get help on correct formatting and usage of filewords.

@d8ahazard @dali99

I'm getting this error after using accompanying text files with description. It halts all progress for me:


`Exception with the stupid image again: [Errno 22] Invalid argument: 'Q:\\AI\\Super_SD_2.0\\stable-diffusion-webui\\models\\dreambooth\\773sloppy\\samples\\breasts cleavage cleavage cutout clothing cutout demon girl demon horns demon tail demon wings full body heart cutout horns medium breasts navel pink eyes pink hair pointy ears short hair simple background solo stirrup legwear tail thighhighs toeless legwear toenail polish toenails toes wings double hair bun 773sloppy 773sloppy 1girl500-0.png'`


Could you please please please tell me what I'm doing wrong? I'm pretty sure its with the formatting of my descriptions.

**Comment by wktra** [comment]
@chakalakasp 
I don't want to be a ditz and hijack your thread for nothing. 

Here's what I've learned. These are direct posts and DMs I've gotten from one or two kind people:

`The filewords file looks like this: 1girl, solo, standing, shiny hair, spiked hair, messy hair, red eyes, long sideburns, sidelocks, short hair, brown hair, black choker, collarbone, small breasts, flat chest, (loli:1.1), bare shoulders, detached sleeves, forest, head tilt, black thighhighs, one bandaged leg`

-

`I use just [filewords] as the instance prompt with and include the token I want to use along with the class at the end of my captions for example on the model I just trained my captions looked like a description of the image, "dconway illustration style" where the token is dconway and the class is illustration style. Class prompt was also just illustration style.`

-

`You'll want to use [filewords] in the instance prompt to tell it to use the captions. Once you do that then it will automatically use txt files first, and resort to image file names if no txt file exists with the same name.`

-

`I would not use underscores since they were removed from training for WD1.3 and will be for 1.4 as well. Other than that, it looks good.`

-

`You can name the class images with whatever you want the classword to be. So something like 1girl (001), 1girl (002), etc, but if you aren't doing specific captioning on them (which I never have) it's probably easier just to use 1girl (or whatever classword you choose) as the class prompt instead of [filewords]. I only use [filewords] for my instance prompt. If your classword isn't in your captioning, make for the instance dataset, make sure to add it along with filewords in the instance prompt so it would become [filewords] 1girl sticking with the example of 1girl as the classword.`
